cJSON.c

```diff
@@ -2410,3 +2410,70 @@ void cJSON_Minify(char *json)
     /* and null-terminate. */
     *into = '\0';
 }
+
+const char *cJSON_GetObjectCstr(const cJSON *object, const char *string)
+{
+	cJSON *cj = cJSON_GetObjectItem(object, string);
+
+	if (!cj || cj->type != cJSON_String || !cj->valuestring) return NULL;
```

Comment:
You have to check for `(cj->type & 0xFF) != cJSON_String`. (Yes, I know, this is kind of horrible!). That is because the type can also contain additional flags such as `cJSON_StringIsConst`.